src/transformers/models/mistral/modeling_mistral.py

```diff
@@ -1197,14 +1259,29 @@ def forward(
         )
 
     def prepare_inputs_for_generation(
```

Comment:
copied from llama here

Reply:
I have add some extra logic regarding sliding window cache(crop the attention mask during decoding phase)